bluesky/utils.py

```diff
@@ -787,3 +788,33 @@ def sanitize_np(val):
             return val.item()
         return val.tolist()
     return val
+
+
+def status_progress_bar(*status_objects, loop):
+    from tqdm import tqdm
+    N = 2  # max decades to divide steps into
+    pbars = {}
+    for st in status_objects:
+        pbar = tqdm()
+        if hasattr(st, 'device'):
+            initial = status.device.position
+            if hasattr(st, 'target'):
+                dx = abs(st.target - initial)
+                pbar.total = dx / 10**(int(floor(log10(dx))) + (1 - N))
+        pbars[st] = pbar
+    sleep_times = itertools.chain((10**i for i in range(-6, -1)),
+                                  itertools.repeat(0.1))
+    while True:
+        # Poll the status objects.
+        if all(st.done for st in status_objects):
+            break
+        for st, pbar in pbars.items():
```

Comment:
should there we a version of this outside of this loop to finish all of the progress bars?